packages/block-library/src/details/edit.js

```diff
@@ -0,0 +1,80 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	InnerBlocks,
+	RichText,
+	InspectorControls,
+} from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+import { ToggleControl } from '@wordpress/components';
+import { useEffect, useRef } from '@wordpress/element';
+import { SPACE } from '@wordpress/keycodes';
+
+export default ( {
+	attributes,
+	className,
+	clientId,
+	isSelected,
+	setAttributes,
+} ) => {
+	const summaryRef = useRef( null );
+
+	useEffect( () => {
+		if ( ! summaryRef.current ) {
+			return;
+		}
+
+		const keyDownListener = ( e ) => {
+			if ( e.keyCode === SPACE ) {
+				e.preventDefault();
+			}
+		};
+
+		const clickListener = ( e ) => e.preventDefault();
```

Comment:
It might be a little more efficient to define these callbacks outside of the component.

Also, a `summary` can also be "clicked" using the <kbd>Enter</kbd> key, so you'll want to `preventDefault()` for that one as well.

Reply:
I just checked, and <kbd>Enter</kbd> seems to be working fine in the current implementation... at least on Chromium, anyway.